snapcraft/main.py

```diff
@@ -110,6 +113,8 @@ def main():
 
     log.configure(log_level=log_level)
 
+    common.set_enable_parallel_builds(not args['--no-parallel-build'])
+
     try:
         commands.load(cmd).main(argv=args['ARGS'])
```

Comment:
I hate this common and its global variables. Could you delay using it? In here you can pass --no-parallel-build to the build main in argv, which is the only who cares about it. Would that make sense, or just make it worse?